src/order/minimal.lean

```diff
@@ -129,3 +129,25 @@ eq_singleton_iff_unique_mem.2 ⟨h.mem_minimals, λ b hb, hb.2 h.1 $ h.2 hb.1⟩
 
 lemma is_greatest.maximals_eq (h : is_greatest s a) : maximals (≤) s = {a} :=
 eq_singleton_iff_unique_mem.2 ⟨h.mem_maximals, λ b hb, hb.2 h.1 $ h.2 hb.1⟩
+
+lemma is_antichain.max_lower_set_of (hs : is_antichain (≤) s) :
+  maximals (≤) {x | ∃ y ∈ s, x ≤ y} = s :=
+begin
+  ext x,
+  simp only [maximals, exists_prop, mem_set_of_eq, forall_exists_index, and_imp, sep_set_of],
+  refine ⟨λ h, exists.elim h.1 (λ y hy, ((h.2 _ hy.1 rfl.le hy.2).symm.subst hy.1)),
+    λ h, ⟨⟨x,h,rfl.le⟩,λ b y hy hby hxy, _⟩⟩,
+  have : x = y := by_contra (λ h_eq, (hs h hy h_eq (hxy.trans hby)).elim),
+  exact hxy.antisymm (this.symm.subst hby),
+end
+
+lemma is_antichain.min_upper_set_of (hS : is_antichain (≤) s) :
+  minimals (≤) {x | ∃ y ∈ s, y ≤ x} = s :=
```

Comment:
`order_dual` doesn't help here?

Reply:
It does now, since I added dualising lemmas. Thanks!